clarity/src/vm/tests/contracts.rs

```diff
@@ -97,51 +97,78 @@ fn test_get_block_info_eval(
 ) {
-    let contracts = [
-        "(define-private (test-func) (get-block-info? time u1))",
-        "(define-private (test-func) (get-block-info? time block-height))",
-        "(define-private (test-func) (get-block-info? time u100000))",
-        "(define-private (test-func) (get-block-info? time (- 1)))",
-        "(define-private (test-func) (get-block-info? time true))",
-        "(define-private (test-func) (get-block-info? header-hash u1))",
-        "(define-private (test-func) (get-block-info? burnchain-header-hash u1))",
-        "(define-private (test-func) (get-block-info? vrf-seed u1))",
+    let mut test_cases = vec![
+        (
+            "case-1",
+            "(define-private (test-func) (get-block-info? time u1))",
+            Ok(Value::none()),
+        ),
+        (
+            "case-2",
+            "(define-private (test-func) (get-block-info? time block-height))",
+            Ok(Value::none()),
+        ),
+        (
+            "case-3",
+            "(define-private (test-func) (get-block-info? time u100000))",
+            Ok(Value::none()),
+        ),
+        (
+            "case-4",
+            "(define-private (test-func) (get-block-info? header-hash u1))",
+            Ok(Value::none()),
+        ),
+        (
+            "case-5",
+            "(define-private (test-func) (get-block-info? burnchain-header-hash u1))",
+            Ok(Value::none()),
+        ),
+        (
+            "case-6",
+            "(define-private (test-func) (get-block-info? vrf-seed u1))",
+            Ok(Value::none()),
+        ),
     ];
 
-    let expected = [
-        Ok(Value::none()),
-        Ok(Value::none()),
-        Ok(Value::none()),
-        Err(CheckErrors::TypeValueError(TypeSignature::UIntType, Value::Int(-1)).into()),
-        Err(CheckErrors::TypeValueError(TypeSignature::UIntType, Value::Bool(true)).into()),
-        Ok(Value::none()),
-        Ok(Value::none()),
-        Ok(Value::none()),
-    ];
+    // test cases where the clarity-wasm analysis pass would fail.
+    if cfg!(not(feature = "clarity-wasm")) {
+        test_cases.extend([
+            (
+                "case-7",
+                "(define-private (test-func) (get-block-info? time (- 1)))",
+                Err(CheckErrors::TypeValueError(TypeSignature::UIntType, Value::Int(-1)).into()),
+            ),
+            (
+                "case-8",
+                "(define-private (test-func) (get-block-info? time true))",
+                Err(CheckErrors::TypeValueError(TypeSignature::UIntType, Value::Bool(true)).into()),
+            ),
+        ]);
+    }
 
-    let mut placeholder_context = ContractContext::new(
+    let placeholder_context = ContractContext::new(
         QualifiedContractIdentifier::transient(),
         ClarityVersion::Clarity2,
     );
 
     let mut owned_env = tl_env_factory.get_env(epoch);
-    for i in 0..contracts.len() {
+    for (case, contract, expected) in test_cases {
```

Comment:
We could have used `enumerate` and drop all the "case-n" in the test cases, but that’s a nitpick.